File: sol/820/820.cpp
struct Solution {
    int minimumLengthEncoding(vector<string>& words) {
        for (auto &w : words){
            ranges::reverse(w);
        }
        ranges::sort(words);
        int res = words.back().size() + 1;
        for (int i = 0; i < words.size() - 1; ++i){
            if (!words[i + 1].starts_with(words[i])){
                res += words[i].size() + 1;
            }
        }
        return res;
    }
};

File: sol/820/820.py
class Solution:
    def minimumLengthEncoding(self, words: List[str]) -> int:
        words = sorted((w[::-1] for w in words))
        return sum(len(words[i]) + 1 for i in range(len(words) - 1) if not words[i + 1].startswith(words[i])) + len(words[-1]) + 1
